prefetch.bs

```diff
@@ -488,61 +611,36 @@ The <dfn>list of sufficiently strict speculative navigation referrer policies</d
             <div class="note">
                 Implementations might also send vendor-specific headers, like Chromium's `` `Purpose` ``/`` `prefetch` ``, Mozilla's `` `X-moz` ``/`` `prefetch` ``, and WebKit's `` `X-Purpose` ``/`` `preview` ``, for compatibility with existing server software. Over time we hope implementers and server software authors will adopt a standard header.
             </div>
-        1. Let |hypotheticalPartitionKey| be the result of [=determining the network partition key=] given |environment|.
-        1. If |originalPartitionKey| is equal to |hypotheticalPartitionKey|, then return "`Blocked`".
-
-            <div class="note">The prefetch would end up in the same partition as the prefetch came from in this case. |environment| represents the environment that would ordinarily be used during navigation fetch.</div>
-        1. If there are [=credentials=] associated with |request|'s [=request/current URL=] and |hypotheticalPartitionKey|, then [=set/append=] |request|'s [=request/current URL=]'s [=url/origin=] to |originsWithConflictingCredentials|.
-        1. If |request|'s [=request/URL=] is not [=potentially trustworthy URL|potentially trustworthy=], then return "`Blocked`".
+        1. Let |proposedPartitionKey| be the result of [=determining the network partition key=] given |request|'s [=request/reserved client=].
+        1. If |partitioningScheme| is "`same-partition`" and |proposedPartitionKey| is not equal to |sourcePartitionKey|, then return "`Blocked`".
+        1. Otherwise, if |partitioningScheme| is "`uncredentialed`" and |proposedPartitionKey| is equal to |sourcePartitionKey|, then return "`Blocked`".
```

Comment:
I think this blocks same-origin prefetches on totally-uncredentialed origins. Is that intentional?